### ISSUE TITLE
Test dance with auto `GOMAXPROCS` build 

### ISSUE THREAD

**Issue by b1ron** [FerretDB/dance#531]
On CI #424 fails with unexpected results for MongoDB. The results differ when run locally. This build FerretDB/FerretDB/pull/3105 has auto `GOMAXPROCS` set and we should test dance with this build to see if it provides consistent results between CI and a local environment. 

Docker run [options](https://docs.docker.com/engine/reference/commandline/run/#options) to set during testing.

**Comment by b1ron** [comment]
Fixed by setting certain ulimits for the service which coincide with recommended [values](https://www.mongodb.com/docs/upcoming/reference/ulimit/#linux-distributions-using-upstart). 

I initially thought it docker compose hitting too many open files and was restarting the service because I when I used `systemd` I encountered this error and setting appropriate ulimit values resolved it. But I could not see this errors in the docker logs.

I also set limits for cpus and memory to match the runner on CI. There's now consistency with CI and a local environment.
